crates/codegen/parser/generator/src/code_generator.rs

```diff
@@ -62,6 +63,18 @@ impl CodeGenerator {
             CargoWorkspace::locate_source_crate("codegen_parser_runtime")?.join("src");
         let mut codegen = Codegen::read_write(&runtime_dir)?;
 
+        {
+            #[derive(Serialize)]
+            pub struct Template<'a> {
+                pub ast_types: &'a AstTypes,
+            }
+            codegen.render(
+                Template { ast_types },
+                runtime_dir.join("napi/templates/ast_backend.rs.jinja2"),
+                output_dir.join("napi/napi_ast_backend.rs"),
```

Comment:
Having a Rust logic named AST backend that is exposed via NAPI and used internally in (expanded) AST frontend in TS is very confusing, at least for me. It very slowly makes sense once I navigate the code more but the naming could probably be improved to better clarify what's happening.

We're already in a compiler setting where frontend/backend meanings are loaded. I don't have any concrete examples yet but I'd love if we simplify this. Arguably, the "backend" could be interpreted as load of helpers? I'll think about it some more.

Reply:
I can change to:

- `ast_model.rs` for the codegen model builder
- `ast_selectors` instead of the backend
- `ast_types` for the front end

WDYT?